### DISCUSSION TITLE
Run projects as a single container

### DISCUSSION THREAD

**Discussion by masaanli** [microsoft/aspire#2068] [General]
I've got a question I've a aspire solution right now with like 30 separate containers.
It works ok, but I also have 30 separate Azure Container Apps now which I don't want.

**2 things I want:**
1. I want to split everything up in code as much as possible, but keep 1 container/application. 
2. When I notice something really needs scaling then i want it to be a separate container.

I'm wanna split things up as much as possible to keep nice bounded contexts, some contexts are very small and are not worth running on a separate container.
So instead of paying for 30 azure container apps, i just not wanna go that way, just wanna go with a few azure app containers or even one. But have everything split up very well. 

That when I wanna split things up, I can do it easily.

So basically:
I want to have 30 api's hosted as 1 application, is that possible trough Aspire?

**Comment by davidfowl** [comment]
> I want to have 30 api's hosted as 1 application, is that possible trough Aspire?

No this isn't something supported out of the box. Maybe split your application into multiple libraries instead? This scenario hasn't come up much but maybe in the future we'll design something (if it turns out to be common).

**Reply by masaanli** [reply]
Thanks for the fast reply.

Yeah, I think i keep sticking for now then with separate libraries ;) Maybe in the future when it's possible I can give it a try again. 

Got some small projects and then it's not worth running with loads of microservices ;)

Are there still benefits of just using Dotnet Aspire with 1 api project then instead of a normal api project?

**Reply by davidfowl** [reply]
Does your one service have any dependencies? Sql, redis? Etc

If so then yes. The telemetry configure is also useful in general

**Reply by masaanli** [reply]
The one now, only got 1 azure sql database that's all.

**Reply by davidfowl** [reply]
So it’ll help a little. The aspire component for sql will wire up resiliency and telemetry.
